providers/https/src/lib.rs

```diff
@@ -171,6 +176,53 @@ async fn check_status(config: Config) -> Result<Blob> {
 }
 
 async fn handle_query(config: Config, request: ProviderRequest) -> Result<Blob> {
+    // HACK: This KeyValueRow structure is a stop-gap measure until we
+    // have ArrayField support in the query data, so we could
+    // directly have Vec of KeyValueRow in the provider request QuerySchema.
+    /// Row of key-value pair with front-end-relevant metadata.
+    /// The metadata needs to stay attached to allow for proper
+    /// synchronization of UI state using only the queryData of
+    /// the provider cell
+    #[derive(Default)]
+    struct KeyValueRow {
+        /// UUID used for React DOM refreshes.
+        /// Safe to ignore here.
+        uuid: String,
+        /// True if the Key-Value pair has been enabled (checked) on UI
+        enabled: bool,
+        /// Key (a header key, or a query param name)
+        key: String,
+        /// Value (a header value, or a query param value)
+        value: String,
+    }
+
+    /// A Row is expected to be serialized as semi-colon separated csv:
+    /// `UUID;BOOL;KEY;VALUE`
+    /// We allow 0/1 or true/false for the bool, but 0/1 is more efficient really
+    impl std::str::FromStr for KeyValueRow {
+        type Err = Error;
+
+        fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+            let mut res = KeyValueRow::default();
+            for (i, val) in s.split(';').enumerate() {
+                match i {
+                    0 => res.uuid = val.into(),
+                    1 => {
+                        if !["1", "0", "true", "false"].contains(&val.to_ascii_lowercase().as_ref())
+                        {
+                            return Err(Error::UnsupportedRequest);
+                        }
+                        res.enabled = val == "1" || &val.to_ascii_lowercase() == "true";
+                    }
+                    2 => res.key = val.into(),
+                    3 => res.value = val.into(),
+                    _ => return Err(Error::UnsupportedRequest),
```

Comment:
This doesn't appear to match the description that additional semicolons in the value are preserved?